dglc/dglc_datamode_noevolve_mod.F90

```diff
@@ -361,31 +370,105 @@ subroutine dglc_datamode_noevolve_advance(pio_subsystem, io_type, io_format, &
             end do
 
             deallocate(lsrf)
-
+            deallocate(usrf)
             call pio_closefile(pioid)
             call pio_freedecomp(pio_subsystem, pio_iodesc)
 
          end do ! end loop over ice sheets
 
       end if
 
+      ! Set initialized flag
+      initialized_noevolve = .true.
+      
       if (initialized_noevolve) then
```

Comment:
With the setting of initialized_noevolve to true at this point, it seems like the `if (initialized_noevolve)` conditional is unnecessary. Do you agree?

Reply:
@billsacks @Katetc - as I pointed out in the comments - I believe that it's important to keep the original ordering if initialize_noevolve. So we still need this.

Reply:
I moved the initialized_noevolve flag setting back to the very last thing that happens in this subroutine.

Reply:
Thanks!